### ISSUE TITLE
Segfault when running test_operator_gpu.test_sparse_dot many times

### ISSUE THREAD

**Issue by haojin2** [apache/incubator-mxnet#11367]
Environment: p2.8xlarge with DL AMI(Ubuntu)
Instruction to reproduce: Build and install MXNet from source with instructions in this link: https://mxnet.incubator.apache.org/install/index.html?platform=Linux&language=Python&processor=GPU, then do "MXNET_TEST_COUNT=10000 nosetests -s --verbose tests/python/gpu/test_operator_gpu.py" and wait.

**Comment by haojin2** [comment]
I already submitted a fix @vrakesh

**Comment by vrakesh** [comment]
@haojin2 ah sounds good :)

**Comment by anirudhacharya** [comment]
SegFault failure log - http://jenkins.mxnet-ci.amazon-ml.com/blue/rest/organizations/jenkins/pipelines/incubator-mxnet/branches/PR-10889/runs/16/nodes/738/log/?start=0

```
test_operator_gpu.test_sparse_dot ... /work/runtime_functions.sh: line 609:     7 Segmentation fault      (core dumped) nosetests-3.4 $NOSE_COVERAGE_ARGUMENTS --with-xunit --xunit-file nosetests_gpu.xml --verbose tests/python/gpu
build.py: 2018-06-26 16:50:33,551 Running of command in container failed (139): nvidia-docker run --rm -t --shm-size=500m -v /home/jenkins_slave/workspace/ut-python3-mkldnn-gpu:/work/mxnet -v /home/jenkins_slave/workspace/ut-python3-mkldnn-gpu/build:/work/build -v /tmp/ci_ccache:/work/ccache -u 1001:1001 -e CCACHE_MAXSIZE=500G -e CCACHE_TEMPDIR=/tmp/ccache -e CCACHE_DIR=/work/ccache -e CCACHE_LOGFILE=/tmp/ccache.log mxnetci/build.ubuntu_gpu /work/runtime_functions.sh unittest_ubuntu_python3_gpu
build.py: 2018-06-26 16:50:33,551 You can try to get into the container by using the following command: nvidia-docker run --rm -t --shm-size=500m -v /home/jenkins_slave/workspace/ut-python3-mkldnn-gpu:/work/mxnet -v /home/jenkins_slave/workspace/ut-python3-mkldnn-gpu/build:/work/build -v /tmp/ci_ccache:/work/ccache -u 1001:1001 -ti --entrypoint /bin/bash -e CCACHE_MAXSIZE=500G -e CCACHE_TEMPDIR=/tmp/ccache -e CCACHE_DIR=/work/ccache -e CCACHE_LOGFILE=/tmp/ccache.log mxnetci/build.ubuntu_gpu /work/runtime_functions.sh unittest_ubuntu_python3_gpu
Traceback (most recent call last):
  File "ci/build.py", line 358, in <module>
    sys.exit(main())
  File "ci/build.py", line 291, in main
    command=command, docker_registry=args.docker_registry, local_ccache_dir=args.ccache_dir)
  File "ci/build.py", line 185, in container_run
    raise subprocess.CalledProcessError(ret, cmd)
```

**Comment by haojin2** [comment]
Fix already submitted in #11389. No new reports, closing this issue.
